File: internal/engine2D/render_object.go
package engine2D

import (
	"linux-wallpaperengine/internal/convert"
	"linux-wallpaperengine/internal/wallpaper"

	rl "github.com/gen2brain/raylib-go/raylib"
)

type RenderObject struct {
	Object         *wallpaper.Object
	Image          *rl.Texture2D
	RenderTexture  *rl.RenderTexture2D
	ParticleSystem *ParticleSystem
	Offset         wallpaper.Vec2
	Cropoffset     wallpaper.Vec2
	Mesh           *convert.MDLMesh
	Effects        []LoadedEffect
	PingPong       [2]*rl.RenderTexture2D
}

File: src/engine/2d/render_object.zig
const std = @import("std");
const sokol = @import("sokol");
const core = @import("core");
const effect = @import("effect.zig");

const sg = sokol.gfx;

pub const RenderObject = struct {
    object_json: std.json.Value,
    image: ?sg.Image = null,
    offset: core.wallpaper.Vec2 = .{},
    effects: []effect.Effect = &.{},
};
